pkg/client/clientset.go

```diff
@@ -53,41 +52,50 @@ func (s ClientSet) Entities() *dtclient.DynatraceClient {
 	return s.dtClient
 }
 
-func CreateClientSet(url string, auth manifest.Auth) (*ClientSet, error) {
-	var dtClient *dtclient.DynatraceClient
-	var autClient *automation.Client
-	var err error
+func CreateClassicClientSet(url string, token string) (*ClientSet, error) {
+	concurrentRequestLimit := environment.GetEnvValueIntLog(environment.ConcurrentRequestsEnvKey)
+
+	dtClient, err := dtclient.NewClassicClient(
+		url,
+		token,
+		dtclient.WithAutoServerVersion(),
+		dtclient.WithClientRequestLimiter(concurrency.NewLimiter(concurrentRequestLimit)),
+	)
+	if err != nil {
+		return nil, err
+	}
 
+	return &ClientSet{
+		dtClient:  dtClient,
+		autClient: nil,
+	}, nil
+}
+
+type PlatformAuth struct {
+	ClientID, ClientSecret, Token string
+}
+
+func CreatePlatformClientSet(url string, auth PlatformAuth) (*ClientSet, error) {
 	concurrentRequestLimit := environment.GetEnvValueIntLog(environment.ConcurrentRequestsEnvKey)
 
-	switch {
-	case auth.OAuth == nil:
-		dtClient, err = dtclient.NewClassicClient(
-			url,
-			auth.Token.Value,
-			dtclient.WithAutoServerVersion(),
-			dtclient.WithClientRequestLimiter(concurrency.NewLimiter(concurrentRequestLimit)),
-		)
-		autClient = nil
-	default:
-		oauthCredentials := clientAuth.OauthCredentials{
-			ClientID:     auth.OAuth.ClientID.Value,
-			ClientSecret: auth.OAuth.ClientSecret.Value,
-			TokenURL:     auth.OAuth.GetTokenEndpointValue(),
-		}
-		dtClient, err = dtclient.NewPlatformClient(
-			url,
-			auth.Token.Value,
-			oauthCredentials,
-			dtclient.WithAutoServerVersion(),
-			dtclient.WithClientRequestLimiter(concurrency.NewLimiter(concurrentRequestLimit)),
-		)
-		autClient = automation.NewClient(
-			url,
-			clientAuth.NewOAuthClient(context.TODO(), oauthCredentials),
-			automation.WithClientRequestLimiter(concurrency.NewLimiter(concurrentRequestLimit)),
-		)
+	oauthCredentials := clientAuth.OauthCredentials{
+		ClientID:     auth.ClientID,
+		ClientSecret: auth.ClientSecret,
+		TokenURL:     auth.Token,
```

Comment:
The OauthCredentials.TokenURL is the URL to get the oauth token for the given client from, isn't it?

It should not be the auth.Token (that's the classic API token right) but it's own value.
See it's initialization in the old version on line 76